assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js

```diff
@@ -318,60 +276,30 @@ describe( 'AudienceTilesWidget', () => {
 			isAudienceSegmentationWidgetHidden: false,
 		} );
 
-		const { container, waitForRegistry } = render(
-			<WidgetWithComponentProps />,
-			{
-				registry,
-			}
-		);
-
-		await waitForRegistry();
-
-		expect( container ).toBeEmptyDOMElement();
-	} );
-
-	it( 'should not render when configuredAudiences is null (not set)', async () => {
-		registry
-			.dispatch( MODULES_ANALYTICS_4 )
-			.setAvailableAudiences( availableAudiences );
-
-		registry.dispatch( CORE_USER ).receiveGetAudienceSettings( {
-			configuredAudiences: null,
-			isAudienceSegmentationWidgetHidden: false,
+		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
+			availableAudiencesLastSyncedAt: Date.now() - 100,
 		} );
 
-		const { container, waitForRegistry } = render(
-			<WidgetWithComponentProps />,
-			{
-				registry,
-			}
-		);
+		await act( async () => {
+			// Render the component and wait for the registry updates
+			const { container, waitForRegistry } = render(
+				<WidgetWithComponentProps />,
+				{
+					registry,
+				}
+			);
 
-		await waitForRegistry();
+			await waitForRegistry();
 
-		expect( container ).toBeEmptyDOMElement();
-	} );
+			// Wait for async updates using waitFor
+			await waitFor( () => {
+				expect( maybeSyncAvailableAudiencesSpy ).toHaveBeenCalledTimes(
+					1
+				);
```

Comment:
I've updated the test locally to use `freezeFetch()`. In my testing it doesn't seem like the `waitFor()` is needed either and the `waitFor()` invocation can simply be replaced with the call to `expect( container ).toMatchSnapshot()`.

Of course I know tests can behave differently in various environments, so it's possible we do still need to use `waitFor()`. Please take another look and see how you fare with it.